fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobManager.java

```diff
@@ -0,0 +1,239 @@
+        boolean registered = registerOverwriteJob(job);
+        if (!registered) {
+            LOG.warn("register insert overwrite job:{} failed", job.getJobId());
+            throw new RuntimeException("register insert overwrite job failed");
+        }
+        try {
+            // get db and table
+            Database database = MetaUtils.getDatabase(context, job.getTargetDbId());
+            OlapTable table = (OlapTable) MetaUtils.getTable(context, database.getId(), job.getTargetTableId());
+            InsertOverwriteJobRunner jobRunner = new InsertOverwriteJobRunner(job, context, stmtExecutor, database, table);
+            jobRunner.run();
+        } finally {
+            deregisterOverwriteJob(job.getJobId());
+        }
+    }
+
+    public boolean registerOverwriteJob(InsertOverwriteJob job) {
+        lock.writeLock().lock();
+        try {
+            if (overwriteJobMap.containsKey(job.getJobId())) {
+                LOG.warn("insert overwrite job:{} is running", job.getJobId());
+                return false;
+            }
+            overwriteJobMap.put(job.getJobId(), job);
+            List<Long> runningPartitions = partitionsWithOverwrite.getOrDefault(job.getTargetTableId(), Lists.newArrayList());
+            if (job.getOriginalTargetPartitionIds() != null) {
+                runningPartitions.addAll(job.getOriginalTargetPartitionIds());
+            }
+            partitionsWithOverwrite.put(job.getTargetTableId(), runningPartitions);
+            jobNum++;
+            return true;
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    public boolean deregisterOverwriteJob(long jobid) {
+        lock.writeLock().lock();
+        try {
+            if (!overwriteJobMap.containsKey(jobid)) {
+                return true;
+            }
+            InsertOverwriteJob job = overwriteJobMap.get(jobid);
+            List<Long> partitionIds = partitionsWithOverwrite.get(job.getTargetTableId());
+            if (partitionIds != null) {
+                partitionIds.removeAll(job.getOriginalTargetPartitionIds());
+                if (partitionIds.isEmpty()) {
+                    partitionsWithOverwrite.remove(job.getTargetTableId());
+                }
+            } else {
+                partitionsWithOverwrite.remove(job.getTargetTableId());
+            }
+            overwriteJobMap.remove(jobid);
+            jobNum--;
+            return true;
+        } catch (Exception e) {
+            LOG.warn("deregister overwrite job failed", e);
+            throw e;
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    public boolean hasRunningOverwriteJob(long tableId) {
+        lock.readLock().lock();
+        try {
+            return partitionsWithOverwrite.containsKey(tableId);
+        } finally {
+            lock.readLock().unlock();
+        }
+    }
+
+    public void replayCreateInsertOverwrite(CreateInsertOverwriteJobInfo jobInfo) {
```

Comment:
log and replay should in the same place.

Reply:
done